src/System.Text.Json/src/System/Text/Json/JsonUtf8Reader.cs

```diff
@@ -0,0 +1,1274 @@
+            }
+            else if (marker == JsonConstants.CloseBrace)
+                EndObject();
+            else if (marker == JsonConstants.CloseBracket)
+                EndArray();
+            else
+                ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.FoundInvalidCharacter, marker);
+            return ConsumeTokenResult.Success;
+        }
+
+        private bool SkipComment()
+        {
+            //Create local copy to avoid bounds checks.
+            ReadOnlySpan<byte> localCopy = _buffer.Slice(_consumed + 1);
+
+            if (localCopy.Length > 0)
+            {
+                byte marker = localCopy[0];
+                if (marker == JsonConstants.Solidus)
+                    return SkipSingleLineComment(localCopy.Slice(1), out _);
+                else if (marker == '*')
+                    return SkipMultiLineComment(localCopy.Slice(1), out _);
+                else
+                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedStartOfValueNotFound, marker);
+            }
+
+            if (IsLastSpan)
+                ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedStartOfValueNotFound, JsonConstants.Solidus);
+            else
+                return false;
+
+            return true;
+        }
+
+        private bool SkipSingleLineComment(ReadOnlySpan<byte> localCopy, out int idx)
+        {
+            //TODO: Match Json.NET's end of comment semantics
+            idx = localCopy.IndexOf(JsonConstants.LineFeed);
+            if (idx == -1)
+            {
+                if (IsLastSpan)
+                {
+                    idx = localCopy.Length;
+                    // Assume everything on this line is a comment and there is no more data.
+                    _lineBytePosition += 2 + localCopy.Length;
+                    goto Done;
+                }
+                else
+                    return false;
+            }
+
+            _consumed++;
+            _lineBytePosition = 0;
+            _lineNumber++;
+        Done:
+            _consumed += 2 + idx;
+            return true;
+        }
+
+        private bool SkipMultiLineComment(ReadOnlySpan<byte> localCopy, out int idx)
+        {
+            idx = 0;
+            while (true)
+            {
+                int foundIdx = localCopy.Slice(idx).IndexOf(JsonConstants.Solidus);
+                if (foundIdx == -1)
+                {
+                    if (IsLastSpan)
+                        ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.EndOfCommentNotFound);
+                    else
+                        return false;
+                }
+                if (foundIdx != 0 && localCopy[foundIdx + idx - 1] == '*')
+                {
+                    idx += foundIdx;
+                    break;
+                }
+                idx += foundIdx + 1;
+            }
+
+            Debug.Assert(idx >= 1);
+            _consumed += 3 + idx;
```

Comment:
Why 3? And why 4 below? (Seriously, I don't understand why those numbers)

Reply:
Will add comments :)
Also, it is `4 - 1` below as well, i.e. 3.

It's adjusting for the `/* */` characters (except the last '*' is already accounted for in idx since idx is before the end '/').